File: ExampleMvvm/librarycommon/src/main/java/com/degde/library/library/login/networking/RetrofitService.java
package com.degde.library.library.login.networking;

import com.degde.library.library.constants.Constantes;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitService {
    private static Retrofit retrofit;

    public static Retrofit getInstanceLogin() {
        if (retrofit == null) {
            retrofit = new Retrofit.Builder().baseUrl(Constantes.URL_SERVER).addConverterFactory(GsonConverterFactory.create()).build();
        }
        return retrofit;
    }

}

File: ExampleMvvm/corekotlin/src/main/java/com/core/mvvm/login/networking/RetrofitService.kt
package com.core.mvvm.login.networking

import com.core.mvvm.costants.Constantes
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitService {
    private var retrofit: Retrofit? = null
    val getInstanceLogin: Retrofit?
        get() {
            if (retrofit == null) {
                retrofit = Retrofit.Builder().baseUrl(Constantes.URL_SERVER).addConverterFactory(GsonConverterFactory.create()).build()
            }
            return retrofit
        }
}